test/integration/etcd_storage_path_test.go

```diff
@@ -0,0 +1,1249 @@
+// it essentially means that you will be break old clusters unless you create some migration path for the old data.
+func TestEtcdStoragePath(t *testing.T) {
+	etcdServer := testutil.RequireEtcd(t)
+	defer testutil.DumpEtcdOnFailure(t)
+	keys := etcd.NewKeysAPI(etcdServer.Client)
+
+	masterConfig, err := testserver.DefaultMasterOptions()
+	if err != nil {
+		t.Fatalf("error getting master config: %#v", err)
+	}
+	masterConfig.AdmissionConfig.PluginOrderOverride = []string{"PodNodeSelector"} // remove most admission checks to make testing easier
+
+	kubeConfigFile, err := testserver.StartConfiguredMaster(masterConfig)
+	if err != nil {
+		t.Fatalf("error starting server: %#v", err)
+	}
+	kubeClient, err := testutil.GetClusterAdminKubeClient(kubeConfigFile)
+	if err != nil {
+		t.Fatalf("error getting client: %#v", err)
+	}
+
+	loader := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(&clientcmd.ClientConfigLoadingRules{ExplicitPath: kubeConfigFile}, &clientcmd.ConfigOverrides{})
+	f := osclientcmd.NewFactory(loader)
+	d := f.Decoder(false)
+	seen := map[reflect.Type]struct{}{}
+
+	if _, err := kubeClient.Core().Namespaces().Create(&kapi.Namespace{ObjectMeta: kapi.ObjectMeta{Name: testNamespace}}); err != nil {
+		t.Fatalf("error creating test namespace: %#v", err)
+	}
+
+	for _, gv := range registered.RegisteredGroupVersions() {
+		for kind, apiType := range kapi.Scheme.KnownTypes(gv) {
+			// we don't care about lists // TODO make sure this is always true
+			if strings.HasSuffix(apiType.Name(), "List") {
+				continue
+			}
+
+			ptrType := reflect.PtrTo(apiType)
+			seen[ptrType] = struct{}{}
+			testData, ok := etcdStorageData[ptrType]
+			pkgPath := apiType.PkgPath()
+
+			if !ok {
+				t.Errorf("no test data for %s from %s.  Please add a test for your new type to etcdStorageData.", kind, pkgPath)
+				continue
+			}
+
+			if testData.ephemeral { // TODO it would be nice if we could remove this and infer if an object is not stored in etcd
+				t.Logf("Skipping test for %s from %s", kind, pkgPath)
+				continue
+			}
+
+			if isInKindAndPathWhiteList(kind, pkgPath) {
+				t.Logf("kind and path are whitelisted: skipping test for %s from %s", kind, pkgPath)
+				continue
+			}
+
+			if len(testData.expectedEtcdPath) == 0 || testData.stub == nil || reflect.TypeOf(testData.stub) != ptrType || isZero(reflect.ValueOf(testData.stub)) {
+				t.Errorf("invalid test data for %s from %s", kind, pkgPath)
+				continue
+			}
+
+			func() { // forces defer to run per iteration of the for loop
+				all := &[]runtime.Object{}
+				defer func() {
+					if !t.Failed() { // do not cleanup if test has already failed since we may need things in the etcd dump
+						if err := cleanup(f, testNamespace, all); err != nil {
+							t.Fatalf("failed to clean up etcd: %#v", err)
+						}
+					}
+					// We create a lot of TCP connections in a tight loop.
+					// This frees them so we do not run out of file descriptors.
+					debug.FreeOSMemory()
```

Comment:
Can you directly nail down which component (openshfit client or etcd client) is causing that problem? Maybe closing and creating a new client would help with that?